src/MissionManager/MissionController.cc

```diff
@@ -360,6 +360,7 @@ VisualMissionItem* MissionController::insertTakeoffItem(QGeoCoordinate /*coordin
 {
     int sequenceNumber = _nextSequenceNumber();
     TakeoffMissionItem * newItem = new TakeoffMissionItem(_controllerVehicle->vtol() ? MAV_CMD_NAV_VTOL_TAKEOFF : MAV_CMD_NAV_TAKEOFF, _masterController, _flyView, _settingsItem, this);
+    _takeoffMissionItem = newItem;
```

Comment:
So it gets set here. But I don't see any code which clears it. The cases where this item goes away needs to be handled and null out the ptr. Also a takeoff item can come in through a plan from the vehicle or one that is loaded from a file. Those cases need to set this as well.

Reply:
@DonLakeFlyer, thanks so much for pointing this out, totally missed it. I just updated the PR, please let me know in case anything else is missing.